### PR TITLE
fix: Platform depended wheels Python versions

### PR DESCRIPTION
**Pull request by jakub-kocka**
The Python version for the wheels build was sometimes incorrect because of the condition.

- fixed the condition
- [fixed `MacOS M1` issue](https://github.com/espressif/idf-python-wheels/pull/30#issuecomment-2158031890)

### CODE REVIEW

**Comment by jakub-kocka** [comment]
Action fixing the incorrect Python version issue:
https://github.com/espressif/idf-python-wheels/actions/runs/9417151712

A new issue with Python 3.12 on MacOS with M1 has occurred.

**Comment by jakub-kocka** [comment]
The issue on` MacOS` with M1 was about `pip` as a Python module. The action uses symlinks to be able to call just `python` for all the versions.

An externally managed environment error was thrown for pip, so it was set to be used from a global scope since no virtual environment is used.

[Successful action fixing this issue](https://github.com/espressif/idf-python-wheels/actions/runs/9446923138/job/26017705359)

File: .github/workflows/build-wheels-python-dependent.yml [hunk 1]
```diff
@@ -60,16 +60,24 @@ jobs:
...
         with:
             python-version: ${{ matrix.python-version }}
 
       - name: Setup Python - MacOS M1
         # Temporary solution until Python version for build will be >= 3.10 (GitHub action support)
-        if: matrix.os == 'macos-latest-xlarge' && matrix.python-version != '3.12'
+        if: matrix.os == 'macos-latest-xlarge'
         run: |
           brew install python@${{ matrix.python-version }}
+          # correcting pip installation for later usage
```
Comment (dobairoland) [inline .github/workflows/build-wheels-python-dependent.yml:73]: @jakub-kocka Thank you for the fix. Can you please explain the reason in these comments? 

```
# correcting pip installation for later usage
# Add the required lines to the pip config file
```

We can see this so these comments are redundant. It is more important to comment why is this necessary. This also helps for the feature to see if the given workaround can be removed because it is not necessary anymore.

Comment (jakub-kocka) [reply .github/workflows/build-wheels-python-dependent.yml]: Thank you @dobairoland, you are right.
I have edited the comments to explain the purpose better.
